### DISCUSSION TITLE
Feature filtering for non-linear correlation

### DISCUSSION THREAD

**Discussion by lilytong20** [blue-yonder/tsfresh#1091] [Q&A]
Hi team, 

Thanks very much for this nice toolbox.  

Can I ask that if the feature filtering process deals with linear relationships between each feature and the target variable only, or non-linear relationship is included too? 

If I want to investigate non-linear significance association between extracted features and the target variables, can tsfresh be used for this? 

Looking forward to some advice. 

Appreciate very much. 

BW,
Lily

**Comment by nils-braun** [comment]
Hi @lilytong20 ! The tsfresh feature selection (tries to) checks for statistical independence of a feature vs. the target and then applies the Benjamini-Yekutieli method to apply this to all features. 

I assume your question relates to a continuous feature and continuous target vector. For this case, we use the Kendal rank test. It does not fully test on statistically independence, but also tests more than just linear correlations. You can find more information in this paper https://arxiv.org/abs/1610.07717 and potentially the references from there.

**Reply by lilytong20** [reply]
Hi,

My question relates to a continuous time series per person, and a binary target. There are in total >700 features extracted from the time series. That was why I'm wondering if only linear associations between each feature and the target are tested. 

Will have a look at the paper. 

Thank you very much for your reply and advice. 

BW,
Lily
